crates/typst-utils/src/format.rs

```diff
@@ -0,0 +1,135 @@
+//! Different formatting functions.
+
+use std::time::Duration;
+
+/// Returns value with `n` digits after floating point where `n` is `precision`.
+/// Standard rounding rule applies (if `n+1`th digit >= 5 then round up).
+/// If `value` is +/- infinity returns `value`.
+///
+/// Will panic for a very large `value` and `precision` (if the `f64` limit is hit).
+/// In practice, it should never panic, otherwise the implementation have to be
+/// more robust.
+pub fn round_with_precision(value: f64, precision: u8) -> f64 {
```

Comment:
In principle I was going to suggest moving this function to `lib.rs`, but, on second thought (after writing my reply to the thread regarding the `assert!` call), I think this function should _only_ be used for formatting purposes, to the point that this should be reflected in the docstring and maybe even in the function's name somehow (possibly unnecessary).

Reply:
Exactly my thoughts! I was also unsure about it's placement at first. Since you would use it through `typst::format` I don't think you have to explain that this function is meant only to format a number. What do you want to add?

Reply:
https://github.com/typst/typst/pull/4587#discussion_r1721075333